scripts/sfpshow

```diff
@@ -350,16 +366,34 @@ def cli():
 @cli.command()
 @click.option('-p', '--port', metavar='<port_name>', help="Display SFP EEPROM data for port <port_name> only")
 @click.option('-d', '--dom', 'dump_dom', is_flag=True, help="Also display Digital Optical Monitoring (DOM) data")
-def eeprom(port, dump_dom):
-    sfp = SFPShow()
-    sfp.display_eeprom(port, dump_dom)
+@click.option('-n', '--namespace', default=None, help="Display interfaces for specific namespace")
+def eeprom(port, dump_dom, namespace):
+    if port and multi_asic.is_multi_asic():
+        ns = multi_asic.get_namespace_for_port(port)
+        if namespace is not None and ns != namespace:
+            print("Error: Interface {} don't belong to this namespace {}".format(port, namespace))
+            return
+        namespace=ns
+
+    sfp = SFPShow(port, namespace, dump_dom)
+    sfp.get_eeprom()
+    sfp.display_eeprom()
 
 # 'presence' subcommand
 @cli.command()
 @click.option('-p', '--port', metavar='<port_name>', help="Display SFP presence for port <port_name> only")
-def presence(port):
-    sfp = SFPShow()
-    sfp.display_presence(port)
+@click.option('-n', '--namespace', default=None, help="Display interfaces for specific namespace")
```

Comment:
can we use the click.choices in the commands in this file  like the commands below ? It will be consistent.

Reply:
click choice() is used in the CLI command handler in show/interfaces/__init.py__. So the user using CLI will get the choice of namespaces. It is not used in the sfpshow script as it was causing issues with the tests.